Update the governance architecture section

mprahl opened the pull request.

mprahl committed:
Update the governance architecture section

Signed-off-by: mprahl <mprahl@users.noreply.github.com>

dockerymick commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,22 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
 View the following component descriptions for the governance architecture diagram:
 
-- *Policy propagator controller:* Evaluates one or more policies on the managed cluster against your specified control and generates Kubernetes events for violations. Violations are propagated to the hub cluster. The following policy controllers are included in your installation, Kubernetes configuration and Certificate. Customize the your policy controllers using advanced configurations.
+- *Policy propagator controller:* Runs on the {product-title-short} hub and is responsible for generating the replicated policies in the managed cluster namespaces on the hub based on the placements bound to the root policy. It also aggregates compliance status from replicated policies to the root policy status. It is also responsible for triggering automations based on policy automations bound t ...
```
I recommend that we say initiating or starting; We avoid using language such as "trigger" for translation.
```suggestion
- *Policy propagator controller:* Runs on the {product-title-short} hub and is responsible for generating the replicated policies in the managed cluster namespaces on the hub based on the placements bound to the root policy. It also aggregates compliance status from replicated policies to the root policy status. It is also responsible for initiating automations based on policy automations bound to the root policy.
```

dockerymick commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,22 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
+- *Policy propagator controller:* Runs on the {product-title-short} hub and is responsible for generating the replicated policies in the managed cluster namespaces on the hub based on the placements bound to the root policy. It also aggregates compliance status from replicated policies to the root policy status. It is also responsible for triggering automations based on policy automations bound t ...
 
-- *Spec sync controller:* Runs on your managed clusters and watches for policy changes to intiate the reconcile process in the cluster namespace of the hub cluster. Every reconcile modifies the `spec` of the replicated policies on the managed cluster to match the `spec` from the hub cluster.
+- *Governance policy addon controller:* Runs on the {product-title-short} hub and is responsible for managing the installation of policy controllers on managed clusters.
```
I have concerns with changing the name here because the image does not show "Governance policy addon controller". 
```suggestion
- *Spec sync controller also known as Governance policy add-on controller:* Runs on the {product-title-short} hub and is responsible for managing the installation of policy controllers on managed clusters.
```

mprahl replied:
It's not changing the name. It just looks that way from the diff. It's actually a separate component just not showed on the image. The spec sync controller is still listed but under the "governance policy framework" section.

mprahl replied:
I'm also hesitant to include this in the image since the image is already very complex. It's just part of the installation flow so it's not as critical that it's in image, however, I think it's useful for the customer to be aware that it exists and what its purpose is.

dockerymick commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,22 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
-* Changes to policies in the watched cluster namespace on the managed cluster.
-* Generated events on policies in the watched cluster namespace on the managed cluster.
-* During each reconcile, the controller creates and updates the policy status for the hub cluster and managed cluster in the cluster namespace, based on the policy changes or generated events. The status only contains updates that are relevant to the current policy and does not consider past statuses if the policy was deleted and recreated.
+- *Governance policy framework:* Runs as the `governance-policy-framework` pod on managed clusters and contains the following controllers.
```
Similar concern that the terms "Governance policy framework" is not labeled in the image. From my perspective, the entire image is the framework. If we move forward with updating the image in 2.11, would it make sense for the framework to have the hub cluster and managed cluster enclosed by a box titled "Governance policy framework? 

```suggestion
- *Governance policy framework:* The previous image represents the framework that runs as the `governance-policy-framework` pod on managed clusters and contains the following controllers:
```

mprahl replied:
It's just grouping the controllers based around the container name they see running on the managed cluster. Each controller within this grouping is listed as bullets underneath which are in the image.

dockerymick commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,22 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
-* Generated events on policies in the watched cluster namespace on the managed cluster.
-* During each reconcile, the controller creates and updates the policy status for the hub cluster and managed cluster in the cluster namespace, based on the policy changes or generated events. The status only contains updates that are relevant to the current policy and does not consider past statuses if the policy was deleted and recreated.
+- *Governance policy framework:* Runs as the `governance-policy-framework` pod on managed clusters and contains the following controllers.
+** *Spec sync controller:* Responsible for synchronizing the replicated policy in the managed cluster namespace on the hub to the managed cluster namespace on the managed cluster.
```
```suggestion
** *Spec sync controller:* Responsible for synchronizing the replicated policy in the managed cluster namespace on the hub cluster to the managed cluster namespace on the managed cluster.
```

dockerymick commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,22 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
+- *Governance policy framework:* Runs as the `governance-policy-framework` pod on managed clusters and contains the following controllers.
+** *Spec sync controller:* Responsible for synchronizing the replicated policy in the managed cluster namespace on the hub to the managed cluster namespace on the managed cluster.
+
+** *Status sync controller:* Responsible for recording compliance events from policy controllers in the replicated policies on the hub and managed cluster. The status only contains updates that are relevant to the current policy and does not consider past statuses if the policy was deleted and recreated.
```
```suggestion
** *Status sync controller:* Responsible for recording compliance events from policy controllers in the replicated policies on the hub and managed cluster. The status only contains updates that are relevant to the current policy and does not consider past statuses if the policy is deleted and recreated.
```

dockerymick commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,22 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
+
+** *Status sync controller:* Responsible for recording compliance events from policy controllers in the replicated policies on the hub and managed cluster. The status only contains updates that are relevant to the current policy and does not consider past statuses if the policy was deleted and recreated.
+
+** *Template sync controller:* Responsible for creating, updating, and deleting objects in the managed cluster namespace on the managed cluster based on the definitions in replicated policy `spec.policy-templates` entries.
```
```suggestion
** *Template sync controller:* Responsible for creating, updating, and deleting objects in the managed cluster namespace on the managed cluster based on the definitions from the replicated policy `spec.policy-templates` entries.
```

dockerymick commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,22 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
 
-- *Template sync controller:* Runs on your managed clusters and watches for template function changes for policies in the cluster namespace on the managed cluster. The changes start the reconcile process, and for each reconcile, it creates,updates, and deletes objects defined in the `spec.policy-templates` of the policies.
+- *Configuration policy controller:* Evaluates configuration policies to inform and enforce desired configuration on managed clusters.
 
-- *Gatekeeper sync controller:* The Gatekeeper sync controller runs on your managed clusters and watches for Gatekeeper constraints in {product-title-short} policies. It then updates the policy status with the Gatekeeper constraint audit results.
+- *Certificate policy controller:* Evaluates certificate policies to inform on certificates stored in `Secret` objects based on organization requirements such as expiring certificates.
```
Similar concerns, the controllers are not listed in the image. I recommend that we do not include them at this point.

mprahl replied:
The configuration policy controller and certificate policy controller are under the "policy controllers" box in the image. If I make them bullets underneath a "Policy Controllers" bullet, would that be sufficient?

dockerymick replied:
I agree that placing a description as part of the Policy controllers. It is not clear that those exact policy controllers are behind the term Policy controllers. I am on the side of not adding the Policy controller descriptions and add a link to the Policy controllers topic from the Additional resources section.

dockerymick reviewed (changes requested):
@mprahl thanks for your updates on the component descriptions. I left some comments

mprahl commented:
@dockerymick great suggestions! Could you please take another look?

dockerymick committed:
Apply suggestions from code review

dockerymick reviewed (approved):
I applied my suggestions and checked the links. This looks good to me, thanks Matt!

mprahl commented:
@dockerymick this looks good to me! Feel free to merge when ready.

jc-berger commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,18 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
 View the following component descriptions for the governance architecture diagram:
 
-- *Policy propagator controller:* Evaluates one or more policies on the managed cluster against your specified control and generates Kubernetes events for violations. Violations are propagated to the hub cluster. The following policy controllers are included in your installation, Kubernetes configuration and Certificate. Customize the your policy controllers using advanced configurations.
+- *Policy propagator controller:* Runs on the {product-title-short} hub and is responsible for generating the replicated policies in the managed cluster namespaces on the hub based on the placements bound to the root policy. It also aggregates compliance status from replicated policies to the root policy status. It is also responsible for initiating automations based on policy automations bound t ...
```
optional:

for consistency of verb tense, we can make both verbs active (runs and generates) & (aggregates and initiates) instead of active and passive (runs and is) & (aggregates and is)

```suggestion
- *Policy propagator controller:* Runs on the {product-title-short} hub and generates the replicated policies in the managed cluster namespaces on the hub based on the placements bound to the root policy. It also aggregates compliance status from replicated policies to the root policy status and initiates automations based on policy automations bound to the root policy.
```

jc-berger commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,18 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
+- *Policy propagator controller:* Runs on the {product-title-short} hub and is responsible for generating the replicated policies in the managed cluster namespaces on the hub based on the placements bound to the root policy. It also aggregates compliance status from replicated policies to the root policy status. It is also responsible for initiating automations based on policy automations bound t ...
 
-- *Spec sync controller:* Runs on your managed clusters and watches for policy changes to intiate the reconcile process in the cluster namespace of the hub cluster. Every reconcile modifies the `spec` of the replicated policies on the managed cluster to match the `spec` from the hub cluster.
+- *Governance policy addon controller:* Runs on the {product-title-short} hub and is responsible for managing the installation of policy controllers on managed clusters.
```
Same as above, and likewise, optional as the original wording is good :) 

```suggestion
- *Governance policy addon controller:* Runs on the {product-title-short} hub and manages the installation of policy controllers on managed clusters.
```

jc-berger commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,18 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
-* Generated events on policies in the watched cluster namespace on the managed cluster.
-* During each reconcile, the controller creates and updates the policy status for the hub cluster and managed cluster in the cluster namespace, based on the policy changes or generated events. The status only contains updates that are relevant to the current policy and does not consider past statuses if the policy was deleted and recreated.
+- *Governance policy framework:* The previous image represents the framework that runs as the `governance-policy-framework` pod on managed clusters and contains the following controllers:
+** *Spec sync controller:* Responsible for synchronizing the replicated policy in the managed cluster namespace on the hub cluster to the managed cluster namespace on the managed cluster.
```
In your previous bullet list, you started each definition with the action that the technology does (runs), so for consistency, I would follow the same sentence structure here:

```suggestion
** *Spec sync controller:* Synchronizes  the replicated policy in the managed cluster namespace on the hub cluster to the managed cluster namespace on the managed cluster.
```

jc-berger commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,18 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
+- *Governance policy framework:* The previous image represents the framework that runs as the `governance-policy-framework` pod on managed clusters and contains the following controllers:
+** *Spec sync controller:* Responsible for synchronizing the replicated policy in the managed cluster namespace on the hub cluster to the managed cluster namespace on the managed cluster.
+
+** *Status sync controller:* Responsible for recording compliance events from policy controllers in the replicated policies on the hub and managed cluster. The status only contains updates that are relevant to the current policy and does not consider past statuses if the policy is deleted and recreated.
```
```suggestion
** *Status sync controller:* Records compliance events from policy controllers in the replicated policies on the hub and managed cluster. The status only contains updates that are relevant to the current policy and does not consider past statuses if the policy is deleted and recreated.
```

jc-berger commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,18 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
+** *Status sync controller:* Responsible for recording compliance events from policy controllers in the replicated policies on the hub and managed cluster. The status only contains updates that are relevant to the current policy and does not consider past statuses if the policy is deleted and recreated.
 
-- *Template sync controller:* Runs on your managed clusters and watches for template function changes for policies in the cluster namespace on the managed cluster. The changes start the reconcile process, and for each reconcile, it creates,updates, and deletes objects defined in the `spec.policy-templates` of the policies.
+** *Template sync controller:* Responsible for creating, updating, and deleting objects in the managed cluster namespace on the managed cluster based on the definitions from the replicated policy `spec.policy-templates` entries.
```
```suggestion
** *Template sync controller:* Creates, updates, and deletes objects in the managed cluster namespace on the managed cluster based on the definitions from the replicated policy `spec.policy-templates` entries.
```

jc-berger commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,18 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
+** *Template sync controller:* Responsible for creating, updating, and deleting objects in the managed cluster namespace on the managed cluster based on the definitions from the replicated policy `spec.policy-templates` entries.
 
-- *Gatekeeper sync controller:* The Gatekeeper sync controller runs on your managed clusters and watches for Gatekeeper constraints in {product-title-short} policies. It then updates the policy status with the Gatekeeper constraint audit results.
+** *Gatekeeper sync controller:* Responsible for recording Gatekeeper constraint audit results as compliance events in corresponding {product-title-short} policies.
```
```suggestion
** *Gatekeeper sync controller:* Records Gatekeeper constraint audit results as compliance events in corresponding {product-title-short} policies.
```

jc-berger reviewed (approved):
/lgtm

dockerymick committed:
Apply suggestions from code review

Great suggestions Jake! Thank you for reminding me about present and active tone.

Co-authored-by: jc-berger <70717303+jc-berger@users.noreply.github.com>

dockerymick commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,18 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
+- *Policy propagator controller:* Runs on the {product-title-short} hub and generates the replicated policies in the managed cluster namespaces on the hub based on the placements bound to the root policy. It also aggregates compliance status from replicated policies to the root policy status and initiates automations based on policy automations bound to the root policy.
 
-- *Spec sync controller:* Runs on your managed clusters and watches for policy changes to intiate the reconcile process in the cluster namespace of the hub cluster. Every reconcile modifies the `spec` of the replicated policies on the managed cluster to match the `spec` from the hub cluster.
+- *Governance policy addon controller:* Runs on the {product-title-short} hub and manages the installation of policy controllers on managed clusters.
```
```suggestion
- *Governance policy add-on controller:* Runs on the {product-title-short} hub cluster and manages the installation of policy controllers on managed clusters.
```

dockerymick committed:
Update governance/grc_arch.adoc

dockerymick commented on governance/grc_arch.adoc:
```diff
@@ -7,19 +7,18 @@ image:../images/governance_arch_2.8.png[Governance architecture diagram]
 View the following component descriptions for the governance architecture diagram:
 
-- *Policy propagator controller:* Evaluates one or more policies on the managed cluster against your specified control and generates Kubernetes events for violations. Violations are propagated to the hub cluster. The following policy controllers are included in your installation, Kubernetes configuration and Certificate. Customize the your policy controllers using advanced configurations.
+- *Policy propagator controller:* Runs on the {product-title-short} hub and generates the replicated policies in the managed cluster namespaces on the hub based on the placements bound to the root policy. It also aggregates compliance status from replicated policies to the root policy status and initiates automations based on policy automations bound to the root policy.
```
```suggestion
- *Policy propagator controller:* Runs on the {product-title-short} hub cluster and generates the replicated policies in the managed cluster namespaces on the hub based on the placements bound to the root policy. It also aggregates compliance status from replicated policies to the root policy status and initiates automations based on policy automations bound to the root policy.
```

dockerymick committed:
Update governance/grc_arch.adoc

dockerymick commented:
@jc-berger thanks for your review and suggestions. I agree and appreciate the reminder about active tone. Thanks @mprahl for initiating the PR, and working through this together. merging this PR
